common/religion/doctrines/wc_doctrines_special.txt

```diff
@@ -164,6 +164,10 @@ doctrine_end_justifies_means = {
 			title_creation_cost_mult = -0.5
 			tyranny_gain_mult = -0.5
 			direct_vassal_opinion = -10
+
+			#Makes hyper loyal units not become mercenaries
+			mercenary_count_mult = -1
```

Comment:
Can't confirm it works. Probably you need to create innovation and use it there.
<details>
<summary>Click to expand</summary>

![image](https://user-images.githubusercontent.com/46384992/144716530-effdb597-2805-493f-a6ae-81a77ded055a.png)

</details>